File: Migratin/Migratin/NSString+Additions.m
//
//  NSString+Additions.m
//  Migratin
//
//  Created by Nichita Litvinov on 20/02/2023.
//

#import "NSString+Additions.h"

@implementation NSString (Additions)

+ (instancetype)stringWithCString:(const char *)cString {
    return [NSString stringWithCString:cString encoding:NSUTF8StringEncoding];
}

- (const char*)toCString {
    if(self == NULL){
        return NULL;
    }
    return strdup([self UTF8String]);
}

@end

File: Migratin/Migratin/NSString+Additions.swift
//
//  NSString+Additions.swift
//  Migratin
//
//  Created by Nichita Litvinov on 16/02/2023.
//

import Foundation

public extension NSString {
    @objc class func stringWithCString(cString: UnsafeMutablePointer<CChar>?) -> NSString? {
        guard let cString = cString else { return nil }
        return NSString(cString: cString, encoding: NSUTF8StringEncoding)
    }
    
    @objc func toCString() -> UnsafeMutablePointer<CChar>? {
        strdup(self.utf8String)
    }
}

